--- a/usr/share/vim/vimfiles/archlinux.vim
+++ b/usr/share/vim/vimfiles/archlinux.vim
@@ -16,12 +16,6 @@
 " These are files we are not likely to want to edit or read.
 set suffixes=.bak,~,.swp,.o,.info,.aux,.log,.dvi,.bbl,.blg,.brf,.cb,.ind,.idx,.ilg,.inx,.out,.toc,.png,.jpg
 
-" Move the swap file location to protect against CVE-2017-1000382
-if ! isdirectory('~/.vim/swap/')
-  silent! call system('install -dm 700 ~/.vim/swap')
-endif
-set directory=~/.vim/swap/
-
 if has('gui_running')
   " Make shift-insert work like in Xterm
   map <S-Insert> <MiddleMouse>
