--- a/dwm.c	2013-10-22 00:30:45.223044585 +0200
+++ b/dwm.c	2013-11-04 18:09:38.816605830 +0100
@@ -2228,6 +2228,8 @@
 	checkotherwm();
 	setup();
 	scan();
+	custom_startup();
 	run();
+	custom_shutdown();
 	cleanup();
 	XCloseDisplay(dpy);
 	return EXIT_SUCCESS;
--- a/config.def.h	2013-10-22 00:30:45.216377919 +0200
+++ b/config.def.h	2013-11-04 18:10:53.099929768 +0100
@@ -50,8 +50,20 @@
 	{ MODKEY|ControlMask|ShiftMask, KEY,       toggletag,      {.ui = 1 << TAG} },
 
 /* helper for spawning shell commands in the pre dwm-5.0 fashion */
-#define SHCMD(cmd) { .v = (const char*[]){ "/bin/sh", "-c", cmd, NULL } }
+#define SHCMD(cmd) &((Arg) { .v = (const char*[]){ "/bin/sh", "-c", cmd, NULL } })
 
+/* startup actions */
+void
+custom_startup(void) {
+	/* place code here */
+}
+
+/* shutdown actions */
+void
+custom_shutdown(void) {
+	/* place code here */
+}
+
 /* commands */
 static const char *dmenucmd[] = { "dmenu_run", "-fn", font, "-nb", normbgcolor, "-nf", normfgcolor, "-sb", selbgcolor, "-sf", selfgcolor, NULL };
 static const char *termcmd[]  = { "uxterm", NULL };
